File: ArraySummation.cpp
#include<iostream>
using namespace std;

class ArrayX
{
    private:
        int *Arr;
        int iSize;

    public:
        ArrayX(int iValue){ //constructor
            this->iSize = iValue;
            Arr = new int[iSize]; // resource allocated
        }

        ~ArrayX(){ // destructor
            delete[] Arr; // resource deallocated
        }

        void Accept(){
            int iCnt = 0; 
            cout<<"Enter the elements"<<endl;
            for(int i = 0; i < iSize;i++){
                cin>>Arr[i];
            }
        }

        void Display(){
            cout<<"Elements of array are: "<<endl;
            for(int i = 0; i < iSize;i++){
                cout<<Arr[i]<<endl;
            }
        }
        int DisplaySum(){
            int iSum =0;
        
            for(int j=0; j<iSize; j++){
                iSum = iSum + Arr[j];
            }
            return iSum;
        }

};

int main(){
    int iRet = 0;

    ArrayX obj1(5);
    obj1.Accept();
    obj1.Display();

    iRet = obj1.DisplaySum();

    cout<<"Summation is:"<<iRet<<endl;

    return 0;
}

File: ArraySummation.java
import java.lang.*;
import java.util.*;

class ArrayX{
    private int Arr[];

    public ArrayX(int iNo){
        Arr = new int[iNo];
    }

    public void Accept(){
        Scanner sobj = new Scanner(System.in);
        System.out.println("Enter the values:");

        for(int iCnt = 0; iCnt < Arr.length; iCnt++){
            Arr[iCnt] = sobj.nextInt();
        }
    }

    public void Display() {
        System.out.println("Array elements are: ");

        for (int iCnt = 0; iCnt < Arr.length; iCnt++) {
            System.out.println(Arr[iCnt]);
        }
    }
    
    public int Summation() {
        int iSum = 0;

        for (int iCnt = 0; iCnt < Arr.length; iCnt++) {
            iSum += Arr[iCnt];
        }
        return iSum;
    }
}

class ArraySummation{
    public static void main(String args[]){
        Scanner sobj = new Scanner(System.in);
        int iLength = 0,iRet=0;

        System.out.println("Enter the value for constructor:");
        iLength = sobj.nextInt();

        ArrayX obj = new ArrayX(iLength);

        obj.Accept();
        obj.Display();

        iRet = obj.Summation();
        System.out.println("Addition is:"+iRet);

        obj = null;
    }
}